core/src/main/java/io/substrait/relation/RelProtoConverter.java

```diff
@@ -374,14 +371,44 @@ public Rel visit(Project project) throws RuntimeException {
             .setCommon(common(project))
             .setInput(toProto(project.getInput()))
             .addAllExpressions(
-                project.getExpressions().stream()
-                    .map(this::toProto)
-                    .collect(java.util.stream.Collectors.toList()));
+                project.getExpressions().stream().map(this::toProto).collect(Collectors.toList()));
 
     project.getExtension().ifPresent(ae -> builder.setAdvancedExtension(ae.toProto()));
     return Rel.newBuilder().setProject(builder).build();
   }
 
+  @Override
+  public Rel visit(Expand expand) throws RuntimeException {
+    var builder =
+        ExpandRel.newBuilder().setCommon(common(expand)).setInput(toProto(expand.getInput()));
+
+    expand
+        .getFields()
+        .forEach(
+            expandField -> {
+              if (expandField instanceof Expand.ConsistentField) {
+                builder.addFields(
+                    ExpandRel.ExpandField.newBuilder()
+                        .setConsistentField(
+                            toProto(((Expand.ConsistentField) expandField).getExpression()))
+                        .build());
+
+              } else {
+                builder.addFields(
+                    ExpandRel.ExpandField.newBuilder()
+                        .setSwitchingField(
+                            ExpandRel.SwitchingField.newBuilder()
+                                .addAllDuplicates(
+                                    ((Expand.SwitchingField) expandField)
+                                        .getDuplicates().stream()
+                                            .map(this::toProto)
+                                            .collect(Collectors.toList())))
+                        .build());
```

Comment:
Can you make an explicit instanceof check for the SwitchingField type here, and then throw an exception on else.

We should have a visitor for ExpandField eventually, but I won't block this PR on it.